packages/core/src/lib/Cache.ts

```diff
@@ -4,158 +4,121 @@ import { ICategory } from "./Category.js"
 import { trace } from "@opentelemetry/api"
 import { Tags } from "../index.js"
 
-type Expiration = { absolute: number } | { relative: number }
-
-export class CacheEntry<State> {
+export class CacheEntry<T> {
   constructor(
     public token: StreamToken,
-    public state: State,
+    public state: T,
+    public cachedAt: number,
   ) {}
 
-  updateIfNewer(supersedes: (a: StreamToken, b: StreamToken) => boolean, other: CacheEntry<State>) {
+  updateIfNewer(other: CacheEntry<T>) {
     if (supersedes(this.token, other.token)) {
       this.token = other.token
       this.state = other.state
+      this.cachedAt = Date.now()
```

Comment:
this needs to move out of the condition, i.e. a re-read needs to freshen the age